### ISSUE TITLE
Launcher and Benchmarker sync more than strictly needed

### ISSUE THREAD

**Issue by anyzelman** [Algebraic-Programming/ALP#244]
The current encapsulation choice in the launcher makes it so that before initialising the ALP management of LPF, that first two memory slots are used to launch (or benchmark) an ALP program. One is used for the LPF collective library, the other is used to register memory areas that are to be broadcast.

However, multiple broadcasts should take place-- currently, for each of them, a separate global registration followed by a synchronisation, in turn followed by the actual broadcast, a de-registration, and another synchronisation. I believe up to three broadcasts may occur (for the benchmarker), which means six synchronisations, while the following solutions could reduce that to four, or even two:

1. use local memory slots and register the related memory areas in one pass;
2. in addition, use auto-coalescing collectives (Suijlen, HLPP, late 2010s I believe).

This is a low priority since it occurs only once per program launch-- if the amount of work is large enough to warrant a multi-node execution, it is almost surely also large enough to hide the overhead of two to four unnecessary synchronisations.
